alchemy/src/upstash/redis.ts

```diff
@@ -2,6 +2,8 @@ import { alchemy } from "../alchemy.js";
 import type { Context } from "../context.js";
 import { Resource } from "../resource.js";
 import type { Secret } from "../secret.js";
+import { UpstashApi } from "./api.js";
```

Comment:
This seems to be a common-enough abstraction to warrant some guidance in `.cursorrules` 👀

Reply:
I think it is in there. Probably not smart enough to remember to put it in its own file.

It's also something we can share across all implementations. The only thing that changes is the auth part